pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/BaseSegmentAssignment.java

```diff
@@ -68,11 +67,14 @@ public abstract class BaseSegmentAssignment implements SegmentAssignment {
   protected String _tableNameWithType;
   protected int _replication;
   protected String _partitionColumn;
+  protected TableConfig _tableConfig;
+  protected Map<InstancePartitionsType, SegmentAssignmentStrategy> _assignmentStrategyMap;
```

Comment:
Looks like not needed anymore

Reply:
we still use this from OfflineSegmentAssignment and RealtimeSegmentAssignment assign and rebalance functions

Reply:
ohh, I missed the _assignmentStrategyMap